### DISCUSSION TITLE
Why does the date/time of a Commit not match the date/time shown under Blame?

### DISCUSSION THREAD

**Discussion by amansaggu26** [community/community#22251] [New to GitHub]
<p>Why does the date/time of a Commit not match the date/time shown under Blame?</p>
<p>For Example:</p>
<p>The Blame shows a Timestamp of (9 Years Ago) on 2011-10-29 10:39:58</p>
<ul>
<li><a href="https://github.com/bitcoin/bips/blame/7e3284dafda168da34888977dbf4a55519b0c54d/bip-0012.mediawiki" rel="noopener nofollow ugc">https://github.com/bitcoin/bips/blame/7e3284dafda168da34888977dbf4a55519b0c54d/bip-0012.mediawiki</a></li>
</ul>
<p>If I then click the Commit itself, the timestamp is 2011-10-29 10:53:18</p>
<ul>
<li><a href="https://github.com/bitcoin/bips/commit/ce40c0f8f02e83892eb185aabea306ee2a3ab10e" rel="noopener nofollow ugc">https://github.com/bitcoin/bips/commit/ce40c0f8f02e83892eb185aabea306ee2a3ab10e</a></li>
</ul>
<p>Why are these timestamps not identical? What is the difference between the two? Any help appreciated :slight_smile:</p>

**Comment by ernest-phillips** [comment]
<p>Hi <a class="mention" href="/amansaggu26">@amansaggu26</a> :wave:</p>
<p>How are you retrieving the dates for these? Please provide the steps so we can reproduce.</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Here is what is likely happening.<br>
A Git commit will contain the name, email address and timestamp of two actors — the author, and the committer.<br>
As per the <a href="https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History">Git - Viewing the Commit History info</a></p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://git-scm.com/favicon.ico" class="site-icon" width="15" height="15">
      <a href="https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History#pretty_format" target="_blank" rel="noopener">git-scm.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="15" height="15">

<h3><a href="https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History#pretty_format" target="_blank" rel="noopener">Git - Viewing the Commit History</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<blockquote>
<p>You may be wondering what the difference is between author and committer. The author is the person who originally wrote the work, whereas the committer is the person who last applied the work. So, if you send in a patch to a project and one of the core members applies the patch, both of you get credit — you as the author, and the core member as the committer.</p>
</blockquote>
<p>The commit view shows when the content was committed, using the committer date.<br>
The blame view shows when the content was authored, using the author date.</p>
<p>You can see both of these dates in full by providing appropriate options to Git log.</p>
<pre><code class="lang-auto">$ git log -1 --pretty=fuller ce40c0f8f02e83892eb185aabea306ee2a3ab10e
commit ce40c0f8f02e83892eb185aabea306ee2a3ab10e
Author:     genjix &lt;fake@lol.u&gt;
AuthorDate: Sat Oct 29 11:39:58 2011 +0100
Commit:     genjix &lt;fake@lol.u&gt;
CommitDate: Sat Oct 29 11:53:18 2011 +0100

    BIP 0001, 0011, 0012, 0013.
</code></pre>

**Comment by amansaggu26** [comment]
<p>Thanks. This is very helpful! :smiley:</p>

**Comment by amansaggu26** [comment]
<p>As a followup question,</p>
<p><strong>The following changes can be seen under blame</strong><br>
ce40c0f8f02e83892eb185aabea306ee2a3ab10e<br>
7c1296b7b62b5a9c09668e1fd66817596d91e6d3<br>
1647c7e6d3e3866454b803fd874df2d0c4991f4a<br>
72f18918a8ae655212ea2700b30f0dc4d758b843<br>
959fecc15bdad070afa63455468b1dba54655fa6</p>
<p><strong>The following changes can be seen under History</strong><br>
1f64d2b06d628dffdc297be9a8afbb9da4e6b6ca<br>
f6db0af39f6eee60ad7a5d8583ad902c139419a6<br>
d9e890a8f27e46806238e298a346397871fd7e87<br>
1647c7e6d3e3866454b803fd874df2d0c4991f4a<br>
72f18918a8ae655212ea2700b30f0dc4d758b843<br>
959fecc15bdad070afa63455468b1dba54655fa6</p>
<p><strong>Why are these two not shown under History?</strong><br>
ce40c0f8f02e83892eb185aabea306ee2a3ab10e<br>
7c1296b7b62b5a9c09668e1fd66817596d91e6d3</p>
<p><strong>Why are these three not shown under Blame?</strong><br>
1f64d2b06d628dffdc297be9a8afbb9da4e6b6ca<br>
f6db0af39f6eee60ad7a5d8583ad902c139419a6<br>
d9e890a8f27e46806238e298a346397871fd7e87</p>
<p>… to say more simply, why does 7c1296b7b62b5a9c09668e1fd66817596d91e6d3 not show up under History? It was authored and committed, yet the commit doesnt appear<br>
</p><aside class="onebox githubcommit">
  <header class="source">
      <a href="https://github.com/bitcoin/bips/commit/7c1296b7b62b5a9c09668e1fd66817596d91e6d3" target="_blank" rel="noopener nofollow ugc">github.com/bitcoin/bips</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/bitcoin/bips/commit/7c1296b7b62b5a9c09668e1fd66817596d91e6d3" target="_blank" rel="noopener nofollow ugc">BIP0012: Accepted -&gt; Withdrawn. To be replaced by new proposal.</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2012-01-03" data-time="21:28:49" data-timezone="UTC">09:28PM - 03 Jan 12 UTC</span>
      </div>

      <div class="user">
        <a href="" rel="noopener">
          <img alt="" src="" class="onebox-avatar-inline" width="20" height="20">
          
        </a>
        
      </div>

      <div class="lines" title="changed 1 files with 1 additions and 1 deletions">
        <a href="https://github.com/bitcoin/bips/commit/7c1296b7b62b5a9c09668e1fd66817596d91e6d3" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+1</span>
          <span class="removed">-1</span>
        </a>
      </div>
    </div>

  </div>
</div>




  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
